### PR TITLE
Migrate HighLighted Cards to Compose ANDROID-11507

### PR DESCRIPTION
**Pull request by afernandezalabarce**
### :tickets: Jira tickets
[Task](https://jira.tid.es/browse/ANDROID-11507)
### :goal_net: What's the goal?
Migrate XML HighLightedCard implementation to Compose

### :construction: How do we do it?
Added new HighLightedCard composable with all settings posibilites made like other compose implementations (settings data classes to minifly the number of parameters.

### ☑️ Checks
- [X] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [ ] Tested with dark mode.
- [X] Tested with API 23.

### :test_tube: How can I test this?
_If it cannot be tested explain why._
- [X] 🖼️ Screenshots/Videos
image

- [ ] Mistica App QR or download link
- [ ] Reviewed by Mistica design team
- [ ] ...

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ buildscript {
         androidx_activity_compose_version = "1.6.0"
         accompanist_version = "0.28.0"
         coil_version = '2.2.2'
+        constraintComposeVersion = '1.0.1'
```
Comment (afernandezalabarce) [inline build.gradle:15]: added constraint layout for compose, amazing for ui design (avoid all columns and rows)

Comment (haynlo) [reply build.gradle:15]: Nice! This makes the compose code much more readable, thanks ^^

Comment (afernandezalabarce) [reply build.gradle:15]: yes! and is very very very powerful!!

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+){
+    val withCustomBackground: Boolean
+        get() = this.imageVector != null || this.bitmap != null || this.drawableResource != null
+}
+
+@Composable
+fun HighLightedCard(
+    modifier: Modifier = Modifier,
+    title: String = "",
+    message: String = "",
```
Comment (haynlo) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:80]: Is this value needed? I think is not used at all

File: README.md [hunk 3]
```diff
@@ -84,6 +84,7 @@ Just set your App or any specific activity to use any of the following:
 * [Tabs](library/src/main/java/com/telefonica/mistica/tabs)
 * [Tags](library/src/main/java/com/telefonica/mistica/tag)
 * [Title](library/src/main/java/com/telefonica/mistica/title)
+* [HighLightedCard (Compose version)](library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard)
```
Comment (haynlo) [inline README.md:87]: I think we can consider to change the structure of this links, in order to group them such as:
* **Highlighted Cards**
        * [XML](library/src/main/java/com/telefonica/mistica/highlightedcard)
        * [Compose](library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard)

Perhaps this is outside the scope of this PR idk 😅

**Review by haynlo — APPROVED** [review]
Nice job! 🙌🏼

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt (hunk 2 above)
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:80]: thanks!  this is a bug... repaired!

File: README.md (hunk 3 above)
Comment (gmerinojimenez) [reply README.md:87]: good idea, can you create a ticket do do it? :)

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/BlauBrand.kt [hunk 4]
```diff
@@ -104,6 +104,7 @@ object BlauBrand : Brand {
         warningHigh = BlauPaletteColor.blau_color_yellow_70,
         warningHighInverse = BlauPaletteColor.blau_color_yellow_70,
         warning = BlauPaletteColor.blau_color_yellow,
+        closeButtonOverlayColor = BlauPaletteColor.blau_color_blue_primary30
```
Comment (pmartinbTEF) [inline library/src/main/java/com/telefonica/mistica/compose/theme/brand/BlauBrand.kt:107]: Why these changes?

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt
Comment (pmartinbTEF) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:1]: Shouldn't be added the corresponding tab to catalog app?

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/BlauBrand.kt (hunk 4 above)
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/theme/brand/BlauBrand.kt:107]: I have based myself on the movistar color palettes, and I have added the missing ones, do you think the change is unnecessary?

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:1]: I didn't see how to add it, and I only created the composable, my apologies!

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt (hunk 2 above)
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt]: > I think "message" should be called "content" like no compose version WDYT?

done!

**Review by pmartinbTEF — APPROVED** [review]
Good job!

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt
Comment (pmartinbTEF) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:1]: Created task for this https://jira.tid.es/browse/ANDROID-13220

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+}
+
+data class HighLightCardCustomBackgroundSettings(
+    val imageVector: ImageVector? = null,
+    val bitmap: ImageBitmap? = null,
+    @DrawableRes val drawableResource: Int? = null,
+){
+    val withCustomBackground: Boolean
+        get() = this.imageVector != null || this.bitmap != null || this.drawableResource != null
+}
```
Comment (gmerinojimenez) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:74]: Minor, but I would put the main composable at the top of the file

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import com.telefonica.mistica.compose.button.Button
+import com.telefonica.mistica.compose.button.ButtonStyle
+import com.telefonica.mistica.compose.theme.MisticaTheme
+
+enum class HighLightCardImageConfig {
+    FIT,
+    FILL,
+    NONE,
```
Comment (gmerinojimenez) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:32]: have you considered [the FIT and FILL](https://www.figma.com/file/tKdPOfcUALzVIh5oizFbm7/%F0%9F%94%B8-Cards-Specs?type=design&node-id=608-2449&t=Z9wYtHYtDusq9hZ2-0) behaviours?

image

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt (hunk 5 above)
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt]: Done!!

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt (hunk 6 above)
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt]: "Yes", 
image

and into settings data class with the property withImage

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/README.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+    val bitmap: ImageBitmap? = null,
+    @DrawableRes val drawableResource: Int? = null,
+){
+    val withCustomBackground: Boolean
+        get() = this.imageVector != null || this.bitmap != null || this.drawableResource != null
+}
+
+```
+
+Its use is very simple, below is a series of composables (Preview) in which we can see each of its uses (the most common)
```
Comment (gmerinojimenez) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/README.md:57]:
```suggestion
Use HighlightedCard composable. There are some examples below
```

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+            }
+        }
+    }
+}
+
+data class HighLightCardImageSettings(
+    val imageVector: ImageVector? = null,
+    val bitmap: ImageBitmap? = null,
+    @DrawableRes val drawableResource: Int? = null,
+    val config: HighLightCardImageConfig = HighLightCardImageConfig.NONE,
```
[the remark this answers is not in the document]
Comment (afernandezalabarce) [reply library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt]: Done!

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+        }
+    }
+}
+
+sealed class HighLightedCardImage{
+    object None: HighLightedCardImage()
+    data class CardBitmap(val bitmap: ImageBitmap): HighLightedCardImage()
+    data class CardImageVector(val imageVector: ImageVector): HighLightedCardImage()
+
+
```
Comment (gmerinojimenez) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:260]: lines

Comment (gmerinojimenez) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:260]:
```suggestion
```

**Review by gmerinojimenez — APPROVED** [review]
Nice job! please remember to add this composable to the catalog
